engineapi/proxy.go

```diff
@@ -80,17 +83,62 @@ func NewEngineClientProxy(im *longhorn.InstanceManager, logger logrus.FieldLogge
 		return nil, err
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	client, err := imclient.NewProxyClient(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort)
+	initProxyTLSClient := func(ip string) (*imclient.ProxyClient, error) {
+		// check for tls cert file presence
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
The implementation creates `ctx` and `cancel` here.
```
ctx, cancel := context.WithCancel(context.Background())
```


Should we pass the `ctx` and `cancel` to this function from the caller instead?
```
func NewEngineClientProxy(ctx context.Context, ctxCancel context.CancelFunc, im *longhorn.InstanceManager, logger logrus.FieldLogger, proxyConnCounter util.Counter)
```

WDYT? @innobead @c3y1huang

Reply:
As already discussed, the ctx should use the parent scope to ensure they are in the same context except for specific reasons.